artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/BindingsImpl.java

```diff
@@ -270,6 +270,11 @@ public Message redistribute(final Message message,
          logger.debug("Message {} being copied as {}", message.getMessageID(), copyRedistribute.getMessageID());
       }
       copyRedistribute.setAddress(message.getAddress());
+      for (SimpleString property : copyRedistribute.getPropertyNames()) {
```

Comment:
@jbertram I think you should rather use coopyRedistribute.clearInternalProperties() here. and if that does not work, there's a predicate thing that Michael Pierce introduce that should have to be initialized.

Reply:
It does work actually....